### DISCUSSION TITLE
MongoDB\Driver\Exception\BulkWriteException: not primary in /app/vendor/mongodb/mongodb/src/Operation/Update.php:209

### DISCUSSION THREAD

**Discussion by masterbater** [jenssegers/laravel-mongodb#2811] [General]
- Laravel-mongodb Version:4.2.0
- PHP Version: 8.3
- Database Driver & Version: 

There is a race problem when using firstOrNew

stack trace
<img width="949" alt="image" src="https://github.com/mongodb/laravel-mongodb/assets/26339988/5114e802-ed50-4aad-9e3d-1ca3ed995ebe">

**Comment by GromNaN** [comment]
Looking at the stack trace, I don't see where you are using firstOrNew.

This error suggests that the command is unable to run on a secondary node because it requires a primary node.

If you use a replica set without load-balancer, you must list all the server hosts in the connection URI.

**Comment by masterbater** [comment]
> Looking at the stack trace, I don't see where you are using firstOrNew.
> 
> This error suggests that the command is unable to run on a secondary node because it requires a primary node.
> 
> If you use a replica set without load-balancer, you must list all the server hosts in the connection URI.

Im using mongodb atlas
